bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/VehicleStateUpdater.java

```diff
@@ -0,0 +1,242 @@
+                        ? OpenClosedType.OPEN
+                        : OpenClosedType.CLOSED);
+        saiCiSMARTHandler.updateState(SAICiSMARTBindingConstants.CHANNEL_DOOR_REAR_RIGHT,
+                chargingStatusResponseMessage.getApplicationData().getBasicVehicleStatus().getRearRightDoor()
+                        ? OpenClosedType.OPEN
+                        : OpenClosedType.CLOSED);
+
+        saiCiSMARTHandler.updateState(SAICiSMARTBindingConstants.CHANNEL_WINDOW_DRIVER,
+                chargingStatusResponseMessage.getApplicationData().getBasicVehicleStatus().getDriverWindow()
+                        ? OpenClosedType.OPEN
+                        : OpenClosedType.CLOSED);
+        saiCiSMARTHandler.updateState(SAICiSMARTBindingConstants.CHANNEL_WINDOW_PASSENGER,
+                chargingStatusResponseMessage.getApplicationData().getBasicVehicleStatus().getPassengerWindow()
+                        ? OpenClosedType.OPEN
+                        : OpenClosedType.CLOSED);
+        saiCiSMARTHandler.updateState(SAICiSMARTBindingConstants.CHANNEL_WINDOW_REAR_LEFT,
+                chargingStatusResponseMessage.getApplicationData().getBasicVehicleStatus().getRearLeftWindow()
+                        ? OpenClosedType.OPEN
+                        : OpenClosedType.CLOSED);
+        saiCiSMARTHandler.updateState(SAICiSMARTBindingConstants.CHANNEL_WINDOW_REAR_RIGHT,
+                chargingStatusResponseMessage.getApplicationData().getBasicVehicleStatus().getRearRightWindow()
+                        ? OpenClosedType.OPEN
+                        : OpenClosedType.CLOSED);
+        saiCiSMARTHandler.updateState(SAICiSMARTBindingConstants.CHANNEL_WINDOW_SUN_ROOF,
+                chargingStatusResponseMessage.getApplicationData().getBasicVehicleStatus().getSunroofStatus()
+                        ? OpenClosedType.OPEN
+                        : OpenClosedType.CLOSED);
+
+        boolean acActive = chargingStatusResponseMessage.getApplicationData().getBasicVehicleStatus()
+                .getRemoteClimateStatus() > 0;
+        saiCiSMARTHandler.updateState(SAICiSMARTBindingConstants.CHANNEL_SWITCH_AC, OnOffType.from(acActive));
+        saiCiSMARTHandler.updateState(SAICiSMARTBindingConstants.CHANNEL_REMOTE_AC_STATUS, new DecimalType(
+                chargingStatusResponseMessage.getApplicationData().getBasicVehicleStatus().getRemoteClimateStatus()));
+
+        saiCiSMARTHandler
+                .updateState(SAICiSMARTBindingConstants.CHANNEL_LAST_POSITION_UPDATE,
+                        new DateTimeType(
+                                ZonedDateTime.ofInstant(
+                                        Instant.ofEpochSecond(chargingStatusResponseMessage.getApplicationData()
+                                                .getGpsPosition().getTimestamp4Short().getSeconds()),
+                                        ZoneId.systemDefault())));
```

Comment:
Was it intentional to get the system timezone, instead of the timezone configured in openHAB?

You might want to use dependency injection (just like you added the httpfactory) to get a hold on `TimeZoneProvider` and use that in the binding for timezonde selection. If you need more details and exmaple, you could look at the astro or airquality binding. 

Please also check other occurences of `ZonedDateTime`

Reply:
I have changed the TimeZone to use the openHAB one.  

I do not think the httpfactory is used correctly.  The implementation uses the "org.eclipse.jetty.client.HttpClient" There is a new version of the API that is a bit more standard HTTP (https://github.com/orgs/SAIC-iSmart-API/discussions/18) that might make implementation with the standard openHAB HTTP Client easier...???? Maybe @tisoft  has some advice about the best way to proceed with this?

Reply:
This comment was about the TimeZoneProvider, and that is now correctly used. 
So there is also an issue with the combination of httpClient and the API? And what is wrong with it? Where do you see difficulties?

Reply:
Sorry I am confusing myself. On 2nd look the httpClient LGTM:
https://github.com/dougculnane/openhab-addons/blob/9b8b04156df33d9a442dc1095e6ec1096af77d20/bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/SAICiSMARTBridgeHandler.java#L264C33-L264C50